### PR TITLE
LPS-173077 Create new DDMExpressionFunction

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
https://issues.liferay.com/browse/LPS-173077

### CODE REVIEW

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -11233,6 +11233,7 @@ old-friendly-urls=Old Friendly URLs
 old-group-name-is-a-required-system-group=Old group name is a required system group.
 old-name=Old Name
 old-role-name-is-a-required-system-role=Old role name is a required system role.
+old-value=Old value
```
Comment (carolmariaabb) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:11236]: @pedro-oliveira446 could you please change to: `Old Value`? All first letters must be capitalized if it is not a sentence, that is, if it does not end with punctuation.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/OldValueFunction.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+public class OldValueFunction
+	implements DDMExpressionFunction.Function1<Object, Object>,
+			   DDMExpressionParameterAccessorAware {
+
+	public static final String NAME = "oldValue";
+
+	@Override
+	public Object apply(Object fieldName) {
+		Map<String, Object> objectEntryOldValues =
+			_ddmExpressionParameterAccessor.getObjectFieldsOldValues();
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/OldValueFunction.java:35]: @pedro-oliveira446 please rename it to `objectFieldsOldValues`

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/util/ObjectCodeEditorUtil.java [hunk 3]
```diff
@@ -222,7 +228,8 @@ public static List<HashMap<String, String>> getItems(Locale locale) {
 			List<HashMap<String, String>> values = new ArrayList<>();
 
 			for (DDMExpressionFunction ddmExpressionFunction : values()) {
-				if (StringUtil.equals(
+				if ((PropsValues.OBJECT_ENTRY_SCRIPT_VARIABLES_VERSION != 2) &&
+					StringUtil.equals(
 						ddmExpressionFunction._key, "old-value") &&
 					!GetterUtil.getBoolean(
 						PropsUtil.get("feature.flag.LPS-171440"))) {
```
Comment (carolmariaabb) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/util/ObjectCodeEditorUtil.java:235]: @pedro-oliveira446 @p-albuquerque if the feature flag is not enabled then we should keep hiding the new feature, even if the version is 2. Could you please double check this logic?

I think it should be something like:

```
StringUtil.equals(ddmExpressionFunction._key, "old-value") &&
(
	!GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-171440")) || 
	(PropsValues.OBJECT_ENTRY_SCRIPT_VARIABLES_VERSION != 2)
)
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Review by p-albuquerque — APPROVED** [review]
LGTM

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
Resent here : https://github.com/brianchandotcom/liferay-portal/pull/129839

**Pull request closed without merging**